runtime/themes/gruvbox.toml

```diff
@@ -1,89 +1,151 @@
+
+"comment" = { fg = "gray", modifiers = ["italic"] }
+"comment.block" = { fg = "gray", modifiers = ["italic"] }
+"comment.block.documentation" = { fg = "gray", modifiers = ["italic"] }
+"comment.line" = { fg = "gray", modifiers = ["italic"] }
+
 "constant" = { fg = "purple1" }
-"constant.builtin" = { fg = "purple1", modifiers = ["bold"] }
-"string" = "green1"
-"constant.numeric" = "purple1"
-"constant.character.escape" = { fg = "fg2", modifiers = ["bold"] }
-"label" = "aqua1"
-"module" = "aqua1"
-
-"diff.plus" = "green1"
-"diff.delta" = "orange1"
-"diff.minus" = "red1"
-
-"warning" = "orange1"
-"error" = "red1"
-"info" = "aqua1"
-"hint" = "blue1"
+"constant.builtin" = { fg = "purple1" }
+"constant.builtin.boolean" = { fg = "purple1" }
+"constant.character" = { fg = "aqua1" }
+"constant.character.escape" = { fg = "orange1" }
+"constant.macro" = { fg = "aqua1" }
+"constant.numeric" = { fg = "purple1" }
+"constructor" = { fg = "purple1" }
+
+"definition" = { underline = { color = "aqua1" } }
+
+"diagnostic" = { underline = { color = "orange1", style = "curl" } }
+"diagnostic.deprecated" = { modifiers = ["crossed_out"] }
+"diagnostic.error" = { underline = { color = "red1", style = "curl" } }
+"diagnostic.hint" = { underline = { color = "aqua1", style = "curl" } }
+"diagnostic.info" = { underline = { color = "yellow1", style = "curl" } }
+"diagnostic.unnecessary" = { modifiers = ["dim"] }
+"diagnostic.warning" = { underline = { color = "orange1", style = "curl" } }
+
+"error" = { fg = "red1" }
+"hint" = { fg = "aqua1" }
+"info" = { fg = "yellow1" }
+"warning" = { fg = "orange1" }
+
+"diff.delta" = { fg = "orange1" }
+"diff.minus" = { fg = "red1" }
+"diff.plus" = { fg = "green1" }
+
+"function" = { fg = "green1" }
+"function.builtin" = { fg = "yellow1" }
+"function.call" = { fg = "green1" }
+"function.macro" = { fg = "blue1" }
+"function.method" = { fg = "green1" }
+
+"keyword" = { fg = "red1" }
+"keyword.control.conditional" = { fg = "red1" }
+"keyword.control.exception" = { fg = "red1" }
+"keyword.control.import" = { fg = "aqua1" }
+"keyword.control.repeat" = { fg = "red1" }
+"keyword.control.return" = { fg = "red1" }
+"keyword.directive" = { fg = "red1" }
+"keyword.function" = { fg = "red1" }
+"keyword.operator" = { fg = "red1" }
+"keyword.storage" = { fg = "red1" }
+"keyword.storage.modifier" = { fg = "red1" }
+"keyword.storage.type" = { fg = "red1" }
+
+"label" = { fg = "red1" }
+
+"markup.bold" = { modifiers = ["bold"] }
+"markup.heading" = "aqua1"
+"markup.italic" = { modifiers = ["italic"] }
+"markup.link.text" = "red1"
+"markup.link.url" = { fg = "green1", modifiers = ["underlined"] }
+"markup.raw" = "red1"
+"markup.strikethrough" = { modifiers = ["crossed_out"] }
+
+"module" = { fg = "aqua1" }
+
+"namespace" = { fg = "fg1" }
+
+"operator" = { fg = "orange1" }
+
+"punctuation" = { fg = "orange1" }
+"punctuation.bracket" = { fg = "orange1" }
+"punctuation.delimiter" = { fg = "orange1" }
+"punctuation.special" = { fg = "orange1" }
```

Comment:
There is no point in specifying more specific identifiers if they match the base group. `punctuation.bracket` will already fall back on `punctuation` which is already `orange1`.

Reply:
Sure! I left only `"punctuation" = { fg = "orange1" }` line :)